lecture-notes/3-bld.md

```diff
@@ -983,17 +985,135 @@ The above diagram with `main.cc` and `square.cc` (C++ files, not C files, but
 same idea) can be compiled two ways, both with the same outcome. The first way
 is as the diagram shows: separate compilation.
 
-```console
+```console?prompt=$
 $ c++ main.cc -c  # compile
 $ c++ square.cc -c  # compile
 $ c++ main.o square.o  # link
 ```
 
+This is the way we have been building programs in our Makefiles throughout
+this module. Now you know what the compilation does but there is still this
+mysterious step that combines object files.
+
+Let's figure out what is going on by experimenting a little bit.
+
+```c
+// main.c
+#include <stdio.h>
+
+extern int random_number();
+
+int main() {
+  printf("Your random number is: %d\n", random_number());
+}
+```
+
+```c
+// lib.c
+int random_number() {
+  return 4;
+}
+```
+
+```console?prompt=$
+$ gcc -Os -c main.c
+$ objdump -M intel --disassemble=main main.o
```

Comment:
Can we do this without `-M intel` for the same reason as above? I know most folks prefer Intel syntax, but I think showing simple commands with "no tricks up our sleeve" is more valuable here.